### ISSUE TITLE
Installation: Warning about inaccessible folders only show up one by one

### ISSUE THREAD

**Issue by jancborchardt** [owncloud/core#24644]
Currently the warnings are:
- inaccessible config folder
- inaccessible apps folder
- inaccessible data folder

These warnings only show up one by one, in that order. The last one even only after you tried to put in your admin username/password and clicking »Finish installation«.

Instead of slowly feeding those out, there should be one page with a list, showing the errors at once. (In case one or more of the folders are already accessible, of course leave those out.)

This will once again simplify the most annoying/technical step of the installation process, which is settings the permissions.

@DeepDiver1975 @PVince81 @schiesbn Is there some reason which led us to do it in these steps? Is it possible to alter so we can streamline a very common installation case?

**Comment by sorvani** [comment]
This is definitely a good add. It would be better served if you could fix SELinux issues in the first place.
On the mangolass.it forum, a bunch of users were complaining that they could not get a CentOS 7 based ownCloud 9.0.X install up so I made instructions for them. 
http://mangolassi.it/topic/9207/install-owncloud-stable-currently-9-0-2-on-centos-7

For a normal install to work, I only needed these SELinux updates. My /data folder already had the rw context.

```
semanage fcontext -a -t httpd_sys_rw_content_t '/var/www/html/owncloud/apps(/.*)?'
restorecon -R /var/www/html/owncloud/apps
semanage fcontext -a -t httpd_sys_rw_content_t '/var/www/html/owncloud/config(/.*)?'
restorecon -R /var/www/html/owncloud/config
```

**Comment by jancborchardt** [comment]
cc @georgehrke because you implemented this partly in https://github.com/owncloud/core/pull/20515 – can you update and also consolidate the warnings?


@sorvani this is not about SELinux specifically but the general installation experience.

**Comment by sorvani** [comment]
@jancborchardt While it may not be specifically about SELinux, that is very commonly the reason for the warning. So why not fix that at the same time as updating the the system to display all the warnings at the same time.

**Comment by PVince81** [comment]
I'd say because fixing the warnings, which can happen in non-SELinux envs, doesn't need the same level of expertise and can be done by anyone while fixing the SELinux part needs more knowledge of SELinux.

In the light of this, it would be good to have a separate ticket to handle the SELinux part.

**Comment by georgehrke** [comment]
The config folder can contain an autoconfig.php, so we actually need the config before we can determine if the actual apps and data folder are read/writable.

**Comment by jancborchardt** [comment]
@georgehrke but the existence of an autoconfig.php can be detected by the webserver without write permission, right? Keep in mind this is all about streamlining the default installation, where there is no autoconfig.

**Comment by georgehrke** [comment]
I think config and apps/data are different things. You can't move the config. The config must be inside config in the ownCloud's root. Having a blocking error message seems reasonable to me. It doesn't make sense to let the user input db name, etc. when config is not read/writable.
In contrast the data directory can be anywhere (and also configurable in the installation dialog).

**Comment by jancborchardt** [comment]
But currently the config and apps errors look the same – blocking single pages. The apps folder error should either be grouped with the config or data one.

**Comment by georgehrke** [comment]
Ok, will check
